crates/ty_python_semantic/src/ast_node_ref.rs

```diff
@@ -52,34 +59,51 @@ impl<T> AstNodeRef<T> {
     /// Dereferencing the `node` can result in undefined behavior if `parsed` isn't the
     /// [`ParsedModuleRef`] to which `node` belongs. It's the caller's responsibility to ensure that
     /// the invariant `node belongs to parsed` is upheld.
-    pub(super) unsafe fn new(parsed: ParsedModuleRef, node: &T) -> Self {
+    pub(super) unsafe fn new(module_ref: &ParsedModuleRef, node: &T) -> Self {
```

Comment:
I think this no longer needs to be unsafe because dereferencing a node index that belongs to another module no longer results in UB but in a panic (out of bounds) or incorrect result. We could probably even add a debug assertion that verifies that `module_ref[index] == node`.

We should also update the comment